### PR TITLE
feat: add resource status summary for environment

### PR DESCRIPTION
**Pull request by aiwantaozi**
**Problem:**

need to show resource/service status in the environment list

**Related Issue:**

#964

### CODE REVIEW

**Comment by gitlawr** [comment]
How does it look with both resource & service now?

**Comment by aiwantaozi** [comment]
> How does it look with both resource & service now?

Should like below, group by ready, transitioning and error
image

File: pkg/apis/environment/basic_view.go [hunk 1]
```diff
@@ -65,9 +68,46 @@ type (
...
 )
 
 type CollectionDeleteRequest = model.EnvironmentDeleteInputs
+
+type environmentOutput struct {
+	model.EnvironmentOutput `json:",inline"`
+	StatusSummary           status.Count `json:"statusSummary"`
+}
+
+func exposeEnvironment(entity *model.Environment) *environmentOutput {
```
Comment (gitlawr) [inline pkg/apis/environment/basic_view.go:85]: Can we move it to pkg/dao/expose.go

File: pkg/apis/environment/basic_view.go [hunk 2]
```diff
@@ -65,9 +68,46 @@ type (
...
+
+	if len(entity.Edges.Resources) > 0 {
+		for _, v := range entity.Edges.Resources {
+			switch {
+			case v.Status.Error:
+				output.StatusSummary.Error++
+			case v.Status.Transitioning:
+				output.StatusSummary.Transitioning++
+			default:
+				output.StatusSummary.Ready++
```
Comment (gitlawr) [inline pkg/apis/environment/basic_view.go:98]: A stopped resource is neither in error nor transitioning status, is it expected to count it as ready?

Comment (aiwantaozi) [reply pkg/apis/environment/basic_view.go:98]: The stopped resource count as "ready" is confusing. What do you think about changing "ready" to "healthy" or "stable"?

File: pkg/apis/environment/basic_view.go (hunk 1 above)
Comment (aiwantaozi) [reply pkg/apis/environment/basic_view.go:85]: `func exposeEnvironment` only changes the expose API response object, and no other places need to use this function, prefer to keep it in the API level, like other expose functions (exposeSetting, exposeVariable)

File: pkg/apis/environment/basic_view.go (hunk 2 above)
Comment (gitlawr) [reply pkg/apis/environment/basic_view.go:98]: Ready, Error and Other?

**Review by gitlawr — APPROVED** [review]
LGTM

**Review by alexcodelf — APPROVED** [review]

**Pull request merged**